core/app/models/spree/wallet.rb

```diff
@@ -61,6 +61,11 @@ def default_wallet_payment_source=(wallet_payment_source)
       raise Unauthorized, "wallet_payment_source #{wallet_payment_source.id} does not belong to wallet of user #{user.id}"
     end
 
+    # Do not update the payment source if the passed source is already default
+    if default_wallet_payment_source.try(:id) == wallet_payment_source.try(:id)
```

Comment:
If both return nil, it should signify that the system is trying to clear the default payment source (wallet_payment_source=>nil) and that the default payment source is already not set (default_wallet_payment_source=>nil). Since we just return for this case, nothing would change, which I would think is what would be wanted.